core/src/block.rs

```diff
@@ -142,6 +142,17 @@ impl BlockBuilder {
                 .collect::<Vec<_>>(),
         );
 
+        // The witness hash of cellbase transaction is assumed to be zero 0x0000....0000
+        let mut witnesses = vec![H256::zero()];
```

Comment:
We may refactor these code to a function of `Block`, e.g. `commit_transactions_ witnesses_root`, it will eliminate duplicate code in builder and verifier, especially hardcoded zero hash for cellbase. And previous code also can be refactored to `commit_transactions_root`.